### ISSUE TITLE
install overwrites existing ~/.config/fish/config.fish

### ISSUE THREAD

**Issue by diggit** [oh-my-fish/oh-my-fish#281]
This is really annoying bug. I had already tweaked `~/.config/fish/config.fish` and installation overwrote it. Not a single question and everything is gone :/

**Comment by coderstephen** [comment]
If everything went properly, the installer should have backed up `~/.config/fish/config.fish` to `~/.config/fish/config.fish.TIMESTAMP.copy`. Double-check if there is a backup put into the config folder. If not, the installer may not have run properly, or maybe has a bug.

**Comment by diggit** [comment]
Oh thanks, there is backup file `config.1458508741.copy`. You saved me!

**Comment by bobthecow** [comment]
The installer _should_ have let you know it was moving that to a backup… I'm curious what might have made that more obvious? Did it just get lost in the pile of things the installer says?

https://github.com/oh-my-fish/oh-my-fish/blob/master/bin/install#L74-L75

**Comment by diggit** [comment]
I had problems with fonts. It made terminal uneasy to read. I am probably too used to new files being renamed to .pcanew and not touching changed ones, from Arch. My fault, sorry for false alarm.

**Comment by JerrySievert** [comment]
i also had this issue, which made me uninstall OMF completely.

it should at least warn that it is overwriting the configuration that we worked hard to get, not just make a copy of it.

this is bad behavior.

**Comment by coderstephen** [comment]
The installer is supposed to print a message indicating if it is going to move the existing config, and where it is putting it. Did the message not work for you?

@JerrySievert What behavior would you prefer? We're open to suggestions on how to improve.

**Comment by JerrySievert** [comment]
@coderstephen i think the best would be to append to the config, not completely overwrite it.

the analogy i can up with during a discussion on irc:

"Welcome back to your room, we painted the walls exactly as you wanted, and we think it looks great!  Oh, we went ahead and rented a storage unit and took all of your furniture to it for you.  We decided that the room would look better if it were empty.  I guess you can move the furniture back in, but we're not going to help you with that."

i did not see the message, it could have been there?  it definitely wasn't obvious.

i uninstalled and moved to `fisher install bobthefish`, and it was a much better and cleaner experience.

**Comment by coderstephen** [comment]
> i did not see the message, it could have been there? it definitely wasn't obvious.

It certainly should have been. Perhaps we could make the text more bold? Or maybe ask for a yes/no confirmation. I'll see if I can direct that to be added to the installer. The last thing we want to do is make users feel like they don't have control over their config files.

Ideally, the bootstrap code that OMF wants to put into `config.fish` should be run _before_ any of your customizations, so that said customizations can use the plugins that you have installed after they are loaded.

> i uninstalled and moved to `fisher install bobthefish`, and it was a much better and cleaner experience.

We're sorry to see you go. Our goal is to continuously improve Oh My Fish! so that it is useful for everyone. We appreciate feedback and constructive criticism.

**Comment by JerrySievert** [comment]
a yes/no confirmation would be great (or even the option to manually merge, or drop the OMF data up front).

i'm here because i was asked to bring it up by @bobthecow in another discussion - and i saw it had already been opened up.

**Comment by coderstephen** [comment]
Those are some good suggestions! I'll see if we can get a timely patch to improve the installer.

I'm thinking something like this sequence:
1. Need to write config.
2. Config already exists.
3. Ask if the user would like to write anyway (moving the old config to a new file name). If yes, continue.
4. If not, print out the "bootstrap" code that OMF needs in order to load properly, and ask the user to add them somewhere to the config file.
5. Continue installation.

What do you think of this?

**Comment by JerrySievert** [comment]
that looks like a great solution!  the only thing i would change would be some comforting message that it's ok to say "no", that you can do it by hand.

i can't speak for everyone, but i kind of feel that someone who is installing a package manager for fish probably knows how to you vi :)

**Comment by coderstephen** [comment]
That's probably a good idea. Friendly messages make installation smoother -- having to install things is already tedious. Thanks for the suggestions, @JerrySievert!

**Comment by JerrySievert** [comment]
good luck! you may win me back at some point :) (though i'm not installing too many things now that i have everything working)

**Comment by brokenbeta** [comment]
The default install behaviour of oh-my-fish is certainly alarming! Not only does the unknown script start running and putting files who knows where, it then claims to backup your old config.fish file -- [but the backup file does not appear](https://cloud.githubusercontent.com/assets/3834867/15825171/96d8f604-2bfa-11e6-972d-7f9f1c37c2fa.png)!

I happened to have the .config/fish folder open in Finder at the time. The backup is simply not there. It didn't work!

Am I in the right folder in Finder? Let's check. Try `ls` in the terminal.

Oh. The file just appeared in Finder. Phew.

Why does it do that? I've tried three times and it's completely repeatable. It's like the file isn't written immediately. It certainly isn't very "nice" behaviour. It definitely dents your confidence in the software when it appears to have "failed" to backup.
